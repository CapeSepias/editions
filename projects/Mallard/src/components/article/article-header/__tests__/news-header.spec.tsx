import React from 'react'
import TestRenderer, { ReactTestRendererJSON } from 'react-test-renderer'
import { NewsHeader } from '..'

describe('NewsHeader', () => {
    it('should show a full featured news header', () => {
        const component: ReactTestRendererJSON | null = TestRenderer.create(
            <NewsHeader
                image={{ source: 'media', path: 'path/to/image.png' }}
                headline="Theresa May makes final speech as PM"
                standfirst="Rolling coverage of the day’s political developments as they happen"
                byline="Andrew Sparrow"
                kicker="News"
            />,
        ).toJSON()
        expect(component).toMatchSnapshot()
    })
    it('should show news header without an image or a kicker', () => {
        const component: ReactTestRendererJSON | null = TestRenderer.create(
            <NewsHeader
                headline="Theresa May makes final speech as PM"
                standfirst="Rolling coverage of the day’s political developments as they happen"
                byline="Andrew Sparrow"
            />,
        ).toJSON()
        expect(component).toMatchSnapshot()
    })
})
