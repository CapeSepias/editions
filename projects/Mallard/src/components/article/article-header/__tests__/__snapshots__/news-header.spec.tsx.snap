// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NewsHeader should show a full featured news header 1`] = `
<View
  style={
    Array [
      Object {
        "alignItems": "flex-start",
        "paddingBottom": 12,
        "paddingHorizontal": 12,
      },
      Object {
        "backgroundColor": "#ffffff",
        "borderColor": "#999999",
      },
    ]
  }
>
  <Image
    source={
      Object {
        "uri": "https://d2cf1ljtg904cv.cloudfront.net/issue/media/phone/media/path/to/image.png",
      }
    }
    style={
      Array [
        Object {
          "backgroundColor": "rgba(0,0,0,.25)",
          "width": "100%",
        },
        Object {
          "aspectRatio": 1.5,
          "marginBottom": 3,
        },
      ]
    }
  />
  <View
    style={
      Array [
        Object {
          "borderBottomWidth": 0.5,
          "marginBottom": 3,
          "paddingBottom": 18,
          "width": "100%",
        },
        Object {
          "backgroundColor": "#ffffff",
          "borderColor": "#999999",
        },
      ]
    }
  >
    <Text
      style={
        Array [
          Object {
            "flexShrink": 0,
            "fontFamily": "GTGuardianTitlepiece-Bold",
            "fontSize": 18,
            "lineHeight": 18,
          },
          Array [
            Object {
              "color": "#121212",
            },
            Object {},
            Object {
              "marginTop": 2.4,
            },
          ],
        ]
      }
    >
      News
    </Text>
  </View>
  <View>
    <Text
      style={
        Array [
          Object {
            "color": "#121212",
            "flexShrink": 0,
            "fontFamily": "GHGuardianHeadline-Regular",
            "fontSize": 28,
            "lineHeight": 30,
          },
          Array [
            Object {
              "marginRight": 24,
              "marginTop": 3,
            },
            Object {
              "color": "#121212",
            },
            Object {},
          ],
        ]
      }
    >
      Theresa May makes final speech as PM
    </Text>
  </View>
  <View>
    <View
      style={
        Array [
          Object {
            "marginBottom": 12,
            "marginTop": 12,
            "paddingTop": 3,
            "width": "100%",
          },
          Object {
            "backgroundColor": "#ffffff",
            "borderColor": "#999999",
          },
          Object {
            "alignContent": "stretch",
            "alignItems": "stretch",
            "justifyContent": "flex-end",
          },
        ]
      }
    >
      <Text
        style={
          Array [
            Object {
              "flexShrink": 0,
              "fontFamily": "GuardianTextEgyptian-Reg",
              "fontSize": 17,
              "lineHeight": 21,
            },
            Array [
              Object {
                "color": "#121212",
              },
              Object {},
            ],
          ]
        }
      >
        Rolling coverage of the day’s political developments as they happen
      </Text>
    </View>
  </View>
  <RNSVGSvgView
    bbHeight={16}
    bbWidth="100%"
    fill="none"
    height={16}
    style={
      Array [
        Object {
          "backgroundColor": "transparent",
          "borderWidth": 0,
        },
        undefined,
        null,
        Object {
          "flex": 0,
          "height": 16,
          "width": "100%",
        },
      ]
    }
    width="100%"
  >
    <RNSVGGroup
      fill={null}
      fillOpacity={1}
      fillRule={1}
      font={Object {}}
      matrix={
        Array [
          1,
          0,
          0,
          1,
          0,
          0,
        ]
      }
      opacity={1}
      originX={0}
      originY={0}
      propList={
        Array [
          "fill",
        ]
      }
      rotation={0}
      scaleX={1}
      scaleY={1}
      skewX={0}
      skewY={0}
      stroke={null}
      strokeDasharray={null}
      strokeDashoffset={null}
      strokeLinecap={0}
      strokeLinejoin={0}
      strokeMiterlimit={4}
      strokeOpacity={1}
      strokeWidth={1}
      vectorEffect={0}
      x={0}
      y={0}
    >
      <RNSVGRect
        fill={
          Array [
            0,
            4288256409,
          ]
        }
        fillOpacity={1}
        fillRule={1}
        height={1}
        matrix={
          Array [
            1,
            0,
            0,
            1,
            0,
            0,
          ]
        }
        opacity={1}
        originX={0}
        originY={0}
        propList={
          Array [
            "fill",
          ]
        }
        rotation={0}
        rx={0}
        ry={0}
        scaleX={1}
        scaleY={1}
        skewX={0}
        skewY={0}
        stroke={null}
        strokeDasharray={null}
        strokeDashoffset={null}
        strokeLinecap={0}
        strokeLinejoin={0}
        strokeMiterlimit={4}
        strokeOpacity={1}
        strokeWidth={1}
        vectorEffect={0}
        width="100%"
        x={0}
        y={0}
      />
      <RNSVGRect
        fill={
          Array [
            0,
            4288256409,
          ]
        }
        fillOpacity={1}
        fillRule={1}
        height={1}
        matrix={
          Array [
            1,
            0,
            0,
            1,
            0,
            0,
          ]
        }
        opacity={1}
        originX={0}
        originY={0}
        propList={
          Array [
            "fill",
          ]
        }
        rotation={0}
        rx={0}
        ry={0}
        scaleX={1}
        scaleY={1}
        skewX={0}
        skewY={0}
        stroke={null}
        strokeDasharray={null}
        strokeDashoffset={null}
        strokeLinecap={0}
        strokeLinejoin={0}
        strokeMiterlimit={4}
        strokeOpacity={1}
        strokeWidth={1}
        vectorEffect={0}
        width="100%"
        x={0}
        y={4}
      />
      <RNSVGRect
        fill={
          Array [
            0,
            4288256409,
          ]
        }
        fillOpacity={1}
        fillRule={1}
        height={1}
        matrix={
          Array [
            1,
            0,
            0,
            1,
            0,
            0,
          ]
        }
        opacity={1}
        originX={0}
        originY={0}
        propList={
          Array [
            "fill",
          ]
        }
        rotation={0}
        rx={0}
        ry={0}
        scaleX={1}
        scaleY={1}
        skewX={0}
        skewY={0}
        stroke={null}
        strokeDasharray={null}
        strokeDashoffset={null}
        strokeLinecap={0}
        strokeLinejoin={0}
        strokeMiterlimit={4}
        strokeOpacity={1}
        strokeWidth={1}
        vectorEffect={0}
        width="100%"
        x={0}
        y={8}
      />
      <RNSVGRect
        fill={
          Array [
            0,
            4288256409,
          ]
        }
        fillOpacity={1}
        fillRule={1}
        height={1}
        matrix={
          Array [
            1,
            0,
            0,
            1,
            0,
            0,
          ]
        }
        opacity={1}
        originX={0}
        originY={0}
        propList={
          Array [
            "fill",
          ]
        }
        rotation={0}
        rx={0}
        ry={0}
        scaleX={1}
        scaleY={1}
        skewX={0}
        skewY={0}
        stroke={null}
        strokeDasharray={null}
        strokeDashoffset={null}
        strokeLinecap={0}
        strokeLinejoin={0}
        strokeMiterlimit={4}
        strokeOpacity={1}
        strokeWidth={1}
        vectorEffect={0}
        width="100%"
        x={0}
        y={12}
      />
    </RNSVGGroup>
  </RNSVGSvgView>
  <Text
    style={
      Array [
        Object {
          "flexShrink": 0,
          "fontFamily": "GuardianTextEgyptian-Bold",
          "fontSize": 17,
          "lineHeight": 21,
        },
        Array [
          Object {
            "color": "#121212",
          },
          Object {},
          Object {
            "marginBottom": 12,
          },
        ],
      ]
    }
  >
    Andrew Sparrow
  </Text>
</View>
`;

exports[`NewsHeader should show news header without an image or a kicker 1`] = `
<View
  style={
    Array [
      Object {
        "alignItems": "flex-start",
        "paddingBottom": 12,
        "paddingHorizontal": 12,
      },
      Object {
        "backgroundColor": "#ffffff",
        "borderColor": "#999999",
      },
    ]
  }
>
  <View>
    <Text
      style={
        Array [
          Object {
            "color": "#121212",
            "flexShrink": 0,
            "fontFamily": "GHGuardianHeadline-Regular",
            "fontSize": 28,
            "lineHeight": 30,
          },
          Array [
            Object {
              "marginRight": 24,
              "marginTop": 3,
            },
            Object {
              "color": "#121212",
            },
            Object {},
          ],
        ]
      }
    >
      Theresa May makes final speech as PM
    </Text>
  </View>
  <View>
    <View
      style={
        Array [
          Object {
            "marginBottom": 12,
            "marginTop": 12,
            "paddingTop": 3,
            "width": "100%",
          },
          Object {
            "backgroundColor": "#ffffff",
            "borderColor": "#999999",
          },
          Object {
            "alignContent": "stretch",
            "alignItems": "stretch",
            "justifyContent": "flex-end",
          },
        ]
      }
    >
      <Text
        style={
          Array [
            Object {
              "flexShrink": 0,
              "fontFamily": "GuardianTextEgyptian-Reg",
              "fontSize": 17,
              "lineHeight": 21,
            },
            Array [
              Object {
                "color": "#121212",
              },
              Object {},
            ],
          ]
        }
      >
        Rolling coverage of the day’s political developments as they happen
      </Text>
    </View>
  </View>
  <RNSVGSvgView
    bbHeight={16}
    bbWidth="100%"
    fill="none"
    height={16}
    style={
      Array [
        Object {
          "backgroundColor": "transparent",
          "borderWidth": 0,
        },
        undefined,
        null,
        Object {
          "flex": 0,
          "height": 16,
          "width": "100%",
        },
      ]
    }
    width="100%"
  >
    <RNSVGGroup
      fill={null}
      fillOpacity={1}
      fillRule={1}
      font={Object {}}
      matrix={
        Array [
          1,
          0,
          0,
          1,
          0,
          0,
        ]
      }
      opacity={1}
      originX={0}
      originY={0}
      propList={
        Array [
          "fill",
        ]
      }
      rotation={0}
      scaleX={1}
      scaleY={1}
      skewX={0}
      skewY={0}
      stroke={null}
      strokeDasharray={null}
      strokeDashoffset={null}
      strokeLinecap={0}
      strokeLinejoin={0}
      strokeMiterlimit={4}
      strokeOpacity={1}
      strokeWidth={1}
      vectorEffect={0}
      x={0}
      y={0}
    >
      <RNSVGRect
        fill={
          Array [
            0,
            4288256409,
          ]
        }
        fillOpacity={1}
        fillRule={1}
        height={1}
        matrix={
          Array [
            1,
            0,
            0,
            1,
            0,
            0,
          ]
        }
        opacity={1}
        originX={0}
        originY={0}
        propList={
          Array [
            "fill",
          ]
        }
        rotation={0}
        rx={0}
        ry={0}
        scaleX={1}
        scaleY={1}
        skewX={0}
        skewY={0}
        stroke={null}
        strokeDasharray={null}
        strokeDashoffset={null}
        strokeLinecap={0}
        strokeLinejoin={0}
        strokeMiterlimit={4}
        strokeOpacity={1}
        strokeWidth={1}
        vectorEffect={0}
        width="100%"
        x={0}
        y={0}
      />
      <RNSVGRect
        fill={
          Array [
            0,
            4288256409,
          ]
        }
        fillOpacity={1}
        fillRule={1}
        height={1}
        matrix={
          Array [
            1,
            0,
            0,
            1,
            0,
            0,
          ]
        }
        opacity={1}
        originX={0}
        originY={0}
        propList={
          Array [
            "fill",
          ]
        }
        rotation={0}
        rx={0}
        ry={0}
        scaleX={1}
        scaleY={1}
        skewX={0}
        skewY={0}
        stroke={null}
        strokeDasharray={null}
        strokeDashoffset={null}
        strokeLinecap={0}
        strokeLinejoin={0}
        strokeMiterlimit={4}
        strokeOpacity={1}
        strokeWidth={1}
        vectorEffect={0}
        width="100%"
        x={0}
        y={4}
      />
      <RNSVGRect
        fill={
          Array [
            0,
            4288256409,
          ]
        }
        fillOpacity={1}
        fillRule={1}
        height={1}
        matrix={
          Array [
            1,
            0,
            0,
            1,
            0,
            0,
          ]
        }
        opacity={1}
        originX={0}
        originY={0}
        propList={
          Array [
            "fill",
          ]
        }
        rotation={0}
        rx={0}
        ry={0}
        scaleX={1}
        scaleY={1}
        skewX={0}
        skewY={0}
        stroke={null}
        strokeDasharray={null}
        strokeDashoffset={null}
        strokeLinecap={0}
        strokeLinejoin={0}
        strokeMiterlimit={4}
        strokeOpacity={1}
        strokeWidth={1}
        vectorEffect={0}
        width="100%"
        x={0}
        y={8}
      />
      <RNSVGRect
        fill={
          Array [
            0,
            4288256409,
          ]
        }
        fillOpacity={1}
        fillRule={1}
        height={1}
        matrix={
          Array [
            1,
            0,
            0,
            1,
            0,
            0,
          ]
        }
        opacity={1}
        originX={0}
        originY={0}
        propList={
          Array [
            "fill",
          ]
        }
        rotation={0}
        rx={0}
        ry={0}
        scaleX={1}
        scaleY={1}
        skewX={0}
        skewY={0}
        stroke={null}
        strokeDasharray={null}
        strokeDashoffset={null}
        strokeLinecap={0}
        strokeLinejoin={0}
        strokeMiterlimit={4}
        strokeOpacity={1}
        strokeWidth={1}
        vectorEffect={0}
        width="100%"
        x={0}
        y={12}
      />
    </RNSVGGroup>
  </RNSVGSvgView>
  <Text
    style={
      Array [
        Object {
          "flexShrink": 0,
          "fontFamily": "GuardianTextEgyptian-Bold",
          "fontSize": 17,
          "lineHeight": 21,
        },
        Array [
          Object {
            "color": "#121212",
          },
          Object {},
          Object {
            "marginBottom": 12,
          },
        ],
      ]
    }
  >
    Andrew Sparrow
  </Text>
</View>
`;
